build_tools/cmake/iree_cc_binary.cmake

```diff
@@ -229,3 +177,74 @@ function(_iree_transitive_dependencies_helper TARGET TRANSITIVE_DEPS)
   # Propagate the augmented list up to the parent scope.
   set(${TRANSITIVE_DEPS} "${_RESULT}" PARENT_SCOPE)
 endfunction()
+
+# Sets target_link_libraries() on all registered binaries.
+# This must be called after all libraries have been declared.
+function(iree_finally_set_binary_link_options)

```

Comment:
nit: maybe 'complete' instead of 'finally' to indicate that it's the completion of a process

Reply:
Sure. Renamed to `iree_complete_binary_link_options()`. I was thinking of try/catch/finally when I choose the name since I wanted it to go all the way at the end :P